src/steps/os/unix.rs

```diff
@@ -49,16 +49,49 @@ pub fn run_homebrew(cleanup: bool, run_type: RunType) -> Result<(), Error> {
     Ok(())
 }
 
+#[cfg(target_os = "linux")]
+#[must_use]
+pub fn run_nix(run_type: RunType) -> Result<(), Error> {
```

Comment:
Isn't this step supposed to run on macOS? What I meant is that only the distribution checking will be performed on Linux. I also rather that the step will be entirely skipped without printing a seperator on nixOS.

You can achieve this by removing `[cfg(target_os = "linux")]` from the function, and instead move the if statement to line 59, before the separator. It should be something like this:

```rust
#[cfg(target_os = "linux")]
{
    if let Ok(Distribution::NixOs) ... {
        debug!("Skipping ...");
        return Err(ErrorKind::SkipStep.into());
    }
}

print_separator("Nix");
# proceed as usual
```

Reply:
Ah, `SkipStep`. Thanks for clarifying!